### PR TITLE
restore-util: add method to set placement rules and store labels

### PR DESCRIPTION
**Pull request by disksing**
Signed-off-by: disksing <i@disksing.com>

### What problem does this PR solve? 
Online restore needs to update PD's placement rules and store labels.

### What is changed and how it works?
- update dependency (tidb+pd)
- add 2 methods

### CODE REVIEW

File: pkg/restore-util/client.go [hunk 1]
```diff
@@ -32,6 +39,13 @@ type Client interface {
 	// ScanRegion gets a list of regions, starts from the region that contains key.
 	// Limit limits the maximum number of regions returned.
 	ScanRegions(ctx context.Context, key, endKey []byte, limit int) ([]*RegionInfo, error)
+	// SetPlacementRule insert or update a placement rule to PD.
```
Comment (lhy1024) [inline pkg/restore-util/client.go:45]:
```suggestion
	// SetPlacementRule inserts or updates a placement rule to PD.
```

File: pkg/restore-util/client.go [hunk 2]
```diff
@@ -32,6 +39,13 @@ type Client interface {
...
+	SetPlacementRule(ctx context.Context, rule Rule) error
+	// DeletePlacementRule removes a placement rule from PD.
+	DeletePlacementRule(ctx context.Context, groupID, ruleID string) error
+	// SetStoreLabel add or update specified label of stores. If labelValue
```
Comment (lhy1024) [inline pkg/restore-util/client.go:49]:
```suggestion
	// SetStoreLabel adds or updates specified label of stores. If labelValue
```

File: pkg/restore-util/client.go (hunk 1 above)
Comment (kennytm) [reply pkg/restore-util/client.go:45]: do you mean to add a suffix -s not -a

**Review by lhy1024 — DISMISSED** [review]
LGTM

File: pkg/restore-util/client.go [hunk 3]
```diff
@@ -181,3 +195,84 @@ func (c *pdClient) ScanRegions(ctx context.Context, key, endKey []byte, limit in
...
+type LabelConstraint struct {
+	Key    string   `json:"key,omitempty"`
+	Op     string   `json:"op,omitempty"`
+	Values []string `json:"values,omitempty"`
+}
+
+// Rule is the placement rule that can be checked against a region. When
+// applying rules (apply means schedule regions to match selected rules), the
+// apply order is defined by the tuple [GroupID, Index, ID].
+type Rule struct {
```
Comment (overvenus) [inline pkg/restore-util/client.go]: Why not use `"github.com/pingcap/pd/server/schedule/placement".Rule`?

File: pkg/restore-util/client.go [hunk 4]
```diff
@@ -181,3 +195,84 @@ func (c *pdClient) ScanRegions(ctx context.Context, key, endKey []byte, limit in
 	}
 	return regionInfos, nil
 }
+
+// LabelConstraint is used to filter store when trying to place peer of a region.
+type LabelConstraint struct {
```
Comment (overvenus) [inline pkg/restore-util/client.go]: Why not use `"github.com/pingcap/pd/server/schedule/placement". LabelConstraint `?

**Review by overvenus — APPROVED** [review]
LGTM

**Pull request merged**